From 6ec816625ff86b559b5ba41ae034bec658f9c431 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 03 Nov 2021 20:00:21 +0200
Subject: [PATCH] [SW-0] gaudi2; arc: modify check for reset source

Currently, reset source value is checked against BIT(HL_COLD_RST)
instead of enum value itself which causes phy firmware copy skip
for cold reset. This patch updates condition to use enum value.

Change-Id: I6663c77c1991d835260b99709f4e2d05682d7379
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index cc2d753..c73f233 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -52,7 +52,7 @@
 	u32 rst_src = hl_rst_src_get();
 
 	/* If not cold reset, dont download PHY FW */
-	if (!(rst_src & BIT(HL_COLD_RST)))
+	if (rst_src != HL_COLD_RST)
 		return;
 
 	for (i = 0; i < (size / 4); i++)
